kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -2726,6 +2731,13 @@ private void updateProducerStats(final TopicPartition topicPartition, final int
                 .getCounter(MESSAGE_IN)
                 .add(numMessages);
 
+        if (conversionCount > 0) {
```

Comment:
Is this check necessary? Could `conversionCount` be negative?

Reply:
This check unnecessary, I have removed it.